### PR TITLE
Python: Update `py/azure-storage/unsafe-client-side-encryption-in-use`

### PR DESCRIPTION
**Pull request by raulgarciamsft**
Update UnsafeUsageOfClientSideEncryptionVersion.ql :
 * predicate `isUnsafeClientSideAzureStorageEncryptionViaObjectCreation` was not useful (it was meant to detect the SDK code, not its usage)
 * fixed & simplified `isUnsafeClientSideAzureStorageEncryptionViaAttributes`, the original query was not finding the right code.

NOTE: tested with a real project: https://github.com/wastore/azure-storage-samples-for-python/tree/master/ClientSideEncryptionToServerSideEncryptionMigrationSamples/ClientSideEncryptionV1ToV2

### CODE REVIEW

File: python/ql/src/experimental/Security/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.ql [hunk 1]
```diff
@@ -16,76 +16,38 @@
...
-    uploadBlob.getACall().asExpr() = call and
-    ctrlFlowNode = call.getAFlowNode() and
-    node.strictlyReaches(ctrlFlowNode) and
-    node != ctrlFlowNode and
+    call.getAChildNode() = attrUploadBlob and
+    node = ctrlFlowNode
+  |
+    s1 in ["key_encryption_key", "key_resolver_function"] and
+    s2 in ["ContainerClient", "BlobClient", "BlobServiceClient"] and
+    s3 in ["upload_blob"] and
```
[the remark this answers is not in the document]
Comment (raulgarciamsft) [reply python/ql/src/experimental/Security/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.ql]: Fixed

**Comment by RasmusWL** [comment]
Thanks for making these updates @raulgarciamsft :+1: I've taken the liberty of suggesting some changes, which you can accept here: https://github.com/raulgarciamsft/ql/pull/32

**Comment by raulgarciamsft** [comment]
> 

Thanks a lot @RasmusWL . I am following up on the PR on my fork that you started as the suggested code seems to have the same flaw than the original (it fails to detect the affected code), hopefully there is a way to integrate your suggestion to fix the query. 

Thanks

**Comment by raulgarciamsft** [comment]
Merged the corrections from @RasmusWL . Thanks a lot.

**Comment by RasmusWL** [comment]
(I had forgotten to update the .expected files after the last commit of https://github.com/raulgarciamsft/ql/pull/32, so I've fixed that directly in this PR)

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**